source/simulator/checkpoint_restart.cc

```diff
@@ -173,37 +152,151 @@ namespace aspect
 #endif
 
     }
+  }
+
+  template<int dim>
+  void Simulator<dim>::log_checkpoint(const std::string filename_for_triangulation, const std::string filename_for_serialization, bool is_quicksave)
+  {
+    std::ofstream checkpoint_log;
```

Comment:
you'll get into trouble here because *every* processor wants to open this file and write into it